fastscanner/tests/test_partitioned_csv_provider.py

```diff
@@ -227,3 +227,62 @@ def test_range_from_key_hour(provider):
     start, end = provider._range_from_key("2023-01", "h")
     assert start.month == 1
     assert end.month == 1
+
+
+@pytest.mark.asyncio
+@patch("fastscanner.adapters.candle.partitioned_csv.ClockRegistry")
+async def test_collect_expired_data_basic(mock_clock_registry, provider):
+    today = datetime(2023, 5, 30, 12, 0, 0)
+    mock_clock_registry.clock.now.return_value = today
+
+    symbol = "AAPL"
+
+    yesterday = today.date() - timedelta(days=1)
+    provider._expirations = {
+        symbol: {
+            "2023-05-22_min": yesterday - timedelta(days=5),  # expired
+            "2023-05-29_min": yesterday + timedelta(days=2),  # not expired
+            "2023-05_h": yesterday - timedelta(days=3),  # expired
+            "2023-01_d": yesterday + timedelta(days=10),  # not expired
+        }
+    }
+
+    provider._cache_freqs = [
+        "1min",
+        "2min",
+        "3min",
+        "5min",
+        "10min",
+        "15min",
+        "1h",
+        "1d",
+    ]
+
+    dummy_df = pd.DataFrame(
+        {"OPEN": [1], "CLOSE": [2]},
+        index=pd.date_range("2023-05-22", periods=1, freq="T"),
+    )
+    provider.get = AsyncMock(return_value=dummy_df)
```

Comment:
Please don't mock the provider's method call. If you mock the methods of the class you're testing, you're not testing anything because the mocks will return whatever you want.

I'm just noticing that this test file is full of mocks of private methods. No need to fix this now but for the record, that's a really bad practice.

You always need to ask yourself: what is it that you're trying to test? In this case, we're trying to test the partitioned_csv adapter. So don't mock any function in the partitioned_csv adapter. The adapter depends on a store, that store can get mocked because we're currently not testing that component, so instead of mocking the provider.get method, mock the provider.store.get method.